app/controllers/manage_account.js

```diff
@@ -0,0 +1,125 @@
+import Ember from "ember";
+import AsyncMixin, { ERROR_STRATEGIES } from "stock/mixins/async";
+import TitleAndLanguageMixin from "stock/mixins/grades_option";
+
+export default Ember.Controller.extend(AsyncMixin, TitleAndLanguageMixin, {
+  user: Ember.computed.alias("model.user"),
+  userService: Ember.inject.service(),
+  printerService: Ember.inject.service(),
+  selectedAdminPrinterId: "",
+  selectedStockPrinterId: "",
+
+  printers: Ember.computed(function() {
+    return this.get("printerService").allAvailablePrinters();
+  }),
+
+  selectedAdminPrinterDisplay: Ember.computed(
+    "model.user.id",
+    "selectedAdminPrinterId",
+    function() {
+      const printerId = this.get("selectedAdminPrinterId");
+      return this.get("userService").getPrinterForUser(
+        this.get("user"),
+        printerId,
+        "admin"
+      );
+    }
+  ),
+
+  selectedStockPrinterDisplay: Ember.computed(
+    "model.user.id",
+    "selectedStockPrinterId",
+    function() {
+      const printerId = this.get("selectedStockPrinterId");
+      return this.get("userService").getPrinterForUser(
+        this.get("user"),
+        printerId,
+        "stock"
+      );
+    }
+  ),
+
+  hideValidationErrors(target) {
+    this.set(`${target.id}InputError`, false);
+    this.set(`${target.id}ValidationError`, false);
+  },
+
+  actions: {
+    updateUserDetails(e) {
+      let value = e.target.value.trim();
+      let isValid;
+
+      if (Object.keys(this.get("user").changedAttributes()).length === 0) {
+        this.hideValidationErrors(e.target);
+        return;
+      }
+
+      switch (e.target.id) {
+        case "firstName":
+          isValid = Boolean(value);
+          break;
+        case "lastName":
+          isValid = Boolean(value);
+          break;
+        case "email":
+          isValid = this.get("userService").checkUserEmailValidity(
+            value,
+            this.get("mobileNumber")
+          );
+          break;
+      }
+
+      if (isValid) {
+        this.runTask(async () => {
+          let user = this.get("user");
+          user.set(e.target.id, value);
+          user.set(
+            "mobile",
+            this.get("userService").getUserMobileWithCode(
+              this.get("user.mobile")
+            )
+          );
+
+          try {
+            await user.save();
+          } catch (e) {
+            this.get("user").rollbackAttributes();
+            throw e;
+          }
+        }, ERROR_STRATEGIES.MODAL);
+        this.hideValidationErrors(e.target);
+      } else {
+        this.get("user").rollbackAttributes();
+        Ember.$(`#${e.target.id}`).focus();
+
+        e.target.value
+          ? this.set(`${e.target.id}ValidationError`, true)
+          : this.set(`${e.target.id}InputError`, true);
+      }
+    },
+
+    changeAdminPrinter(value) {
```

Comment:
Can we have a common function like
UpdatePrinter(event){
//since API call is same and we can pass tag as a param or make use of _event_ value and get that
}

Reply:
It is already a common API call added in `printer-service` to update printer based on tag of the application.